### PR TITLE
Add HSA functions from ROCR Runtime for HIP 6.1.0

### PR DESCRIPTION
**Pull request by matyas-streamhpc**

### CODE REVIEW

**Comment by matyas-streamhpc** [comment]
Waiting for ROCR Runtime fix. - done

**Comment by matyas-streamhpc** [comment]
Rebasing is required when https://github.com/ROCm/HIP/pull/3445 is merged.

File: docs/sphinx/_toc.yml.in [hunk 1]
```diff
@@ -15,6 +15,7 @@ subtrees:
 
 - caption: Reference
   entries:
+<<<<<<< HEAD
```
Comment (yhuiYH) [inline docs/sphinx/_toc.yml.in]: TOC needs to be resolved.

Comment (matyas-streamhpc) [reply docs/sphinx/_toc.yml.in]: Thanks, resolved.

Comment (yhuiYH) [reply docs/sphinx/_toc.yml.in]: Thanks... @matyas-streamhpc can you double check why the build failed now?  I tried to rebuild it on readthedocs

Comment (matyas-streamhpc) [reply docs/sphinx/_toc.yml.in]: It is a recurring issue. I do not really know the reason, but I have an intuition that something is cashed and it makes a conflict. I wanted to open a ticket to investigate, but I forgot it. It happens with the original version too. I think I should take a look onto the CI policy, but that is just my first impression.

I did not modify anything just I pushed the commit with a new hash number and the CI works.

**Review by gandryey — APPROVED** [review]

**Comment by matyas-streamhpc** [comment]
The PR was rebased and conflicts are resolved again. Please take a look at it!

File: docs/index.md [hunk 2]
```diff
@@ -19,6 +19,7 @@ portable applications for AMD and NVIDIA GPUs from single source code.
 
 * {doc}`./understand/programming_model`
 * {doc}`./understand/programming_model_reference`
+* {doc}`./reference/virtual_rocr`
```
Comment (yhuiYH) [inline docs/index.md]: If you add it to Reference, please remove from "Conceptual" index card.  We don't want 2 entries on the index page.

**Review by yhuiYH — CHANGES_REQUESTED** [review]

File: docs/index.md (hunk 2 above)
Comment (matyas-streamhpc) [reply docs/index.md]: Nice catch, I pushed the fix.

Comment (yhuiYH) [reply docs/index.md]: Looks better.  Thx for the fix.

**Review by yhuiYH — APPROVED** [review]

**Pull request merged**